salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
It does not produce the same output in a multi-master environment. Your solution would produce this file (the same in all the workers):

```
10.10.0.2 api api.infra.caasp.local a0848146a8854c519ce698d28901e824 a0848146a8854c519ce698d28901e824.caasp.infra.local
10.10.0.5 api api.infra.caasp.local f9848a36ae854c519ce398d25907e828 f9848a36ae854c519ce398d25907e828.caasp.infra.local
```

while my solution would produce:

```
10.10.0.2 api api.infra.caasp.local
10.10.0.2 a0848146a8854c519ce698d28901e824 a0848146a8854c519ce698d28901e824.caasp.infra.local
10.10.0.5 f9848a36ae854c519ce398d25907e828 f9848a36ae854c519ce398d25907e828.caasp.infra.local
```

with `api.infra.caasp.local` being `10.10.0.2` **or** `10.0.0.5`